repository/src/lib.rs

```diff
@@ -316,11 +316,93 @@ impl<T: RawRepository> DistributedRepository<T> {
         unimplemented!()
     }
 
-    /// Creates an agenda commit on top of the `work` branch.
+    /// Creates an extra-agenda transaction commit on top of the `work` branch.
     pub async fn create_extra_agenda_transaction(
         &mut self,
-        _transaction: &ExtraAgendaTransaction,
+        transaction: &ExtraAgendaTransaction,
     ) -> Result<CommitHash, Error> {
-        unimplemented!()
+        let work_commit = self.raw.locate_branch(WORK_BRANCH_NAME.into()).await?;
+        let last_header_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
```

Comment:
```suggestion
        let last_header_commit_hash = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
```
This is hash value not commit, so I suggest you about using work_commit_hash and last_header_commit_hash

Reply:
maybe we need to make rules for this because there are lots of code mixed up using commit and commit hash as the same meaning.

let me think about it for a bit.